File: An 2/Sem 1/MAP/Lab9/C#/Lab2/UI/StudentsObserver.cs
using System;
using Lab2.Lab2_Controller;
using System.Collections;
using Lab2.Lab2_Model;
using System.Collections.Generic;

namespace Lab2
{
	namespace Lab2_UI {
		public class StudentsObserver : IObserver<IDictionary<int, Student>>
		{
			private String comparator;

			private IDisposable unsubscriber;
			public StudentsObserver (String comparator)
			{
				this.comparator = comparator;
			}

			public virtual void Subscribe(Controller provider) {
				this.unsubscriber = provider.Subscribe (this);
			}

			public virtual void OnNext(IDictionary<int, Student> arg) {
				foreach (Student student in arg.Values) {
					if ("<".Equals(this.comparator)) {
						if (student.average() < 5) {
							Console.WriteLine (student.ToString() + "< 5");
						}
					} else {
						if (student.average() >= 5) {
							Console.WriteLine (student.ToString() + ">= 5");
						}
					}
				}
			}

			public virtual void OnCompleted() {

			}

			public virtual void OnError(Exception e) {

			}

			public virtual void Unsubscribe() {
				this.unsubscriber.Dispose ();
			}
		}
	}
}

File: An 2/Sem 2/AOP/Lab 3/src/com/company/UI/StudentsObserver.java
package com.company.UI;

import com.company.Model.Student;

import java.util.Map;
import java.util.Observable;
import java.util.Observer;

public class StudentsObserver implements Observer {
    private String comparator;

    public StudentsObserver(String comparator) {
        this.comparator = comparator;
    }

    @Override
    public void update(Observable o, Object arg) {
        Map<Integer, Student> map = (Map<Integer, Student>) arg;
        for (Student student : map.values()) {
            if ("<".equals(this.comparator)) {
                if (student.average() < 5) {
                    System.out.println(student.toString() + "< 5");
                }
            } else {
                if (student.average() >= 5) {
                    System.out.println(student.toString() + ">= 5");
                }
            }
        }
    }
}
